examples/translation_example.py

```diff
@@ -64,75 +70,83 @@ def _forward(self, batch):
             max_length=self.config.max_length,
             num_beams=self.config.num_beams,
             num_return_sequences=1,
+            repetition_penalty=1.2,
         )
 
     def forward(self, batch):
-        if self.autocast:
-            with torch.autocast(device_type="cuda"):
-                outputs = self._forward(batch)
-        else:
-            outputs = self._forward(batch)
+        outputs = self._forward(batch)
         return outputs
 
 
 class ModelForSeq2SeqModel(HFModel):
     def __init__(self, config):
-        self.config = config
-        super().__init__(config.pretrained_model_name_or_path)
+        self.trans_config = config
+        self.config = AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=self.trans_config.pretrained_model_name_or_path,
+            trust_remote_code=True,
+        )
+        super().__init__(self.trans_config.pretrained_model_name_or_path)
 
     def load_model(self, device="cuda"):
         model = CustomModel(
-            self.config.pretrained_model_name_or_path, self.config.autocast
+            self.trans_config, self.trans_config.pretrained_model_name_or_path, self.trans_config.autocast
         )
         model = model.to(device)
+        model.half()
         model.eval()
         return model
 
     def load_config(self):
         return AutoConfig.from_pretrained(
-            pretrained_model_name_or_path=self.config.pretrained_model_name_or_path,
+            pretrained_model_name_or_path=self.trans_config.pretrained_model_name_or_path,
             trust_remote_code=True,
         )
 
     @lru_cache(maxsize=1)
     def load_tokenizer(self):
         return AutoTokenizer.from_pretrained(
-            pretrained_model_name_or_path=self.config.pretrained_model_name_or_path,
+            pretrained_model_name_or_path=self.trans_config.pretrained_model_name_or_path,
             trust_remote_code=True,
         )
 
     def max_seq_length(self) -> int:
-        return self.config.max_length
+        return self.config.max_source_positions
 
     @lru_cache(maxsize=1)
     def load_cfg(self):
-        return AutoConfig.from_pretrained(
-            pretrained_model_name_or_path=self.config.pretrained_model_name_or_path,
+        config = AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=self.trans_config.pretrained_model_name_or_path,
             trust_remote_code=True,
         )
-
+        return config
 
 def preprocess_df(df):
     ip = load_object_on_worker("IndicProcessor", IndicProcessor, {"inference": True})
-    sentences = df["text"].to_arrow().to_pylist()
+    indices = df['indic_proc_text'].index.to_arrow().to_pylist()
```

Comment:
I think you can keep index on GPU 
```suggestion
    indices = df['indic_proc_text'].index
```

Reply:
Done